Source/WebKit/UIProcess/wpe/WebPageProxyWPE.cpp

```diff
@@ -152,11 +155,13 @@ Vector<RendererBufferFormat> WebPageProxy::preferredBufferFormats() const
             break;
         }
 
+#if USE(GBM)
         WPEDRMDevice* targetDevice = wpe_buffer_formats_get_group_device(formats, i);
         if (!targetDevice)
             targetDevice = mainDevice;
```

Comment:
```suggestion
            targetDevice = wpe_buffer_buffer_formats_get_device(formats);
```

and add it here. Unless the diff confused me, it doesn't seem like `mainDevice` is used elsewhere, and `formats` is sed in a const-way (although that's not explicit. Should we?)

This allows us to remove the guards earlier.

Reply:
The idea is to use the `wpe_buffer_formats_get_device()` getter only once before the loop... although, given that it is a simple getter, maybe it's fine to move the call around as you suggest.

Reply:
Sorry! I totally missed this was in a loop: in that case, I have no strong opinion :pray:

Reply:
I left it outside of the loop in the end.